File: src/test/resources/generic_base.lua
--- @module generic_base

--[[- 
@source src/test/java/cc/tweaked/javadoc/files/GenericBase.java:25
@type generic_base
]]
local generic_base = {}

--[[- Method on the concrete subclass.

@source src/test/java/cc/tweaked/javadoc/files/GenericBase.java:31
@treturn number Always 1.
]]
function generic_base.getOwn() end

--[[- Inherited getter, declared on the generic base.

@source src/test/java/cc/tweaked/javadoc/files/GenericBase.java:16
@treturn number Always 42.
]]
function generic_base.getInherited() end

File: src/test/java/cc/tweaked/javadoc/files/GenericBase.java
package cc.tweaked.javadoc.files;

import dan200.computercraft.api.lua.LuaFunction;

public class GenericBase {
    /**
     * Abstract base with a free type parameter. Its @LuaFunctions should
     * still attach to concrete @cc.module subclasses below.
     */
    public static abstract class Base<T> {
        /**
         * Inherited getter, declared on the generic base.
         *
         * @return Always 42.
         */
        @LuaFunction
        public final int getInherited() {
            return 42;
        }
    }

    /**
     * @cc.module generic_base
     */
    public static class Concrete extends Base<String> {
        /**
         * Method on the concrete subclass.
         *
         * @return Always 1.
         */
        @LuaFunction
        public final int getOwn() {
            return 1;
        }
    }
}
